jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1696,6 +1696,8 @@ void java_bytecode_parsert::rinner_classes_attribute(
     // This is a marker that a class is anonymous.
     if(inner_name_index == 0)
       parsed_class.is_anonymous_class = true;
+    else
+      parsed_class.inner_name = pool_entry_lambda(inner_name_index).s;
```

Comment:
:pick: `pool_entry_lambda` is not a good name - I assumed it was related to Java lambdas rather than it is a C++ lambda.  In this case - you can call `pool_entry` directly. But you might like to rename it to `pool_entry_lookuper`

Reply:
:question: So the `name` on an inner class is `A$B` and the inner_name is `B`?

Reply:
Wait hang on - we're inside a loop here, if there are multiple inner classes which one gets the inner_name?

Reply:
The only case where the loop does not hit the `continue;` is the one where the entry is found for the inner class we are currently parsing. There are no cross-class updates in this function.